packages/forms/src/components/BasicForm.tsx

```diff
@@ -98,10 +87,22 @@ function BasicForm<TFieldValues extends FieldValues>({
       // Clear the cache as well (no longer needed)
       removeFromSessionStorage(cacheKey);
     }
+    setShowErrorSummary(false);
     // Fire the submit we passed in
     return onSubmit(data);
   };
 
+  const handleInvalidSubmit: SubmitErrorHandler<TFieldValues> = (errors) => {
+    const flatErrors = flattenErrors(errors);
+    setShowErrorSummary(flatErrors.length > 0);
+
+    errorSummaryRef.current?.focus();
+  };
+
+  React.useEffect(() => {
+    errorSummaryRef.current?.focus();
+  }, [showErrorSummary]);
```

Comment:
Why do we do it this way, where we set `showErrorSummary` state, and wait for that to trigger a useEffect hook? Why not replace `setShowErrorSummary(true)` with `errorSummaryRef.current?.focus();`? 

(Or more likely, with a `focusErrorSummary` function which does the focusing.)

Reply:
Nevermind. I tried removing this useEffect since I thought this would work without it, but I see its necessary for focusing to work the first time.

Reply:
Yeah. I tried really hard to not need it but sadly it seems like it is required 😓